Scale hide value label

rugheid opened the pull request:
This pull request adds an optional boolean to all scale answer formats to hide the value label of the scale. The boolean value is not included in the initialisers, but has to be set explicitly if one wants to override the default behaviour. Two steps are added to ORKCatalog to the scale task. The first added step contains a discrete horizontal scale with a hidden value label. The second added step contains a continuous vertical scale with a hidden value label.

rsanchezsaez commented:
I have not reviewed the code, but I suggest you move the new scale test cases from `ORKCatalog` to `ORKTest`.

The general idea behind `ORKCatalog` is to keep it a light and simple catalog of the different question types, rather than an exhaustive one of all their available options.

rugheid commented:
Ok, I understand! I'll move it.

rugheid commented:
I removed the extra steps from ORKCatalog and I think there isn't actually anything to add to te tests. So I guess this pull request is ready for review.

YuanZhu-apple commented:
@rugheid Thanks for this PR!

Can you give some use cases of this feature? Do you have concerns about current value shown to users could affect their answer?

rugheid commented:
@YuanZhu-apple I worked with a couple of doctors on my last project. They all agreed that showing the user a number would affect the results. When a user just sees a bar, he/she will have to choose depending on his/her feeling. The doctors thought that otherwise the user would start thinking in numbers and just set the bar to the number he/she had in mind.
This applies best to the continuous scale, but I can imagine some developers to remove the label for the discrete scale and text scale as well.

YuanZhu-apple commented on ResearchKit/Common/ORKAnswerFormat.h:
```diff
@@ -477,6 +482,11 @@ ORK_CLASS_AVAILABLE
+/**
+ A Boolean value indicating whether the value label should be hidden. (read-only)
+ */
+@property (assign, getter=shouldHideValueLabel) BOOL hideValueLabel;
```
Would `hideSelectedValue` be a better name?

YuanZhu-apple commented on ResearchKit/Common/ORKScaleSliderView.m:
```diff
@@ -423,6 +431,14 @@ - (instancetype)initWithFormatProvider:(id<ORKScaleAnswerFormatProvider>)formatP
+            
+            if ([_formatProvider shouldHideValueLabel]) {
+                [self addConstraints:
+                 [NSLayoutConstraint constraintsWithVisualFormat:@"V:[_valueLabel(==0)]"
```
I have seen this constraint twice, can you consolidate them? Maybe merge the latest changes from master branch first.

YuanZhu-apple commented:
@rugheid Your point is valid. Thanks for the explanation.

YuanZhu-apple commented:
Can you also update `ORKESerialization`, which handles the JSON encoding/decoding.

rugheid commented:
@YuanZhu-apple I changed the name of the property to `hideSelectedValue`, updated `ORKESerialization` and changed `ORKScaleSliderView` to use the constraints only once. All changes from master were already merged.

YuanZhu-apple commented:
Thanks! 
Can you also resolve the conflicts?

umerkhan-apple commented:
@rugheid Were you able to make time to resolve the conflicts?

rugheid commented:
@umerkhan-apple I think I resolved most conflicts (not yet pushed though). I did not have time yet to test properly. I'll try to get it done this weekend.

umerkhan-apple commented:
@rugheid Awesome, sounds good.

umerkhan-apple commented:
@rugheid Any updates on this PR?

rugheid commented:
@umerkhan-apple I did not find the time yet to update it, sorry...

umerkhan-apple commented:
@rugheid No worries, let me know if you need any help. Also let me know whenever you've update it, so I can do a final review.

rugheid commented:
@umerkhan-apple Will do! Thanks

pmitche commented:
What's the status of this PR?

rugheid commented:
I've not been able to fix the conflicts yet. If you don't want to keep this pull request open, I'll open a new one once I've fixed the conflicts.
Sorry for taking so long.

YuanZhu-apple commented:
ok, closed for now.
